src/org/etools/j1939_84/controllers/part02/Part02Step17Controller.java

```diff
@@ -584,16 +609,22 @@ private void testSp12730(OBDModuleInformation module) {
                     + module.getModuleName() + " for PG "
                     + GHG_TRACKING_LIFETIME_PG);
         } else {
+            var partOnePacket = get(packetForPg.getPgnDefinition().getId(), module.getSourceAddress(), 1);
             packetForPg.getSpns()
                        .forEach(spn -> {
                            // 6.2.17.12.b. Fail PG query where any bin value received is greater than FAFFh.
                            if (spn.getRawValue() > 0xFAFFFFFFL) {
                                addFailure("6.2.17.12.b - Bin value received is greater than 0xFAFFFFFFL(h) from "
                                        + module.getModuleName() + " for " + spn);
                            }
-                           // FIXME: this needs to be implemented on the dataRepo bug is fixed
                            // 6.2.17.12.c - Fail all values where the corresponding value received in part 1 is
                            // greater than the part 2 value (where supported)
+                           var partOneValue = partOnePacket.getSpnValue(spn.getId()).orElse(NOT_AVAILABLE);
+                           if (partOneValue > spn.getRawValue()) {
```

Comment:
This comparison (and others) is wrong.  You're comparing an engineering value with a raw value.